pkg/util/cmd/encoder.go

```diff
@@ -74,10 +78,8 @@ func (c *tidbEncoder) encodeError(f zapcore.Field) {
 		}
 	}
 }
-func (c *tidbEncoder) EncodeEntry(ent zapcore.Entry, fields []zapcore.Field) (*buffer.Buffer, error) {
-	c.line.Free()
-	c.line = _pool.Get()
-
+func (e *tidbEncoder) EncodeEntry(ent zapcore.Entry, fields []zapcore.Field) (*buffer.Buffer, error) {
+	c := e.clone()
```

Comment:
In
https://github.com/uber-go/zap/blob/4a895a245ab794e3ae965783843a9fbc793083bc/zapcore/json_encoder.go#L365
and
https://github.com/pingcap/log/blob/master/zap_text_encoder.go#L406
they are both `e.clone()`. So you modified it?